realtime/redisSubscriber.js

```diff
@@ -22,14 +23,31 @@ const featureToggles = require('feature-toggles');
  */
 module.exports = (io) => {
   sub.on('message', (channel, mssgStr) => {
-    // message object to be sent to the clients
-    const mssgObj = JSON.parse(mssgStr);
-    const key = Object.keys(mssgObj)[0];
+    try {
+      const mStr = Buffer.from(mssgStr, 'hex');
+      unzipAsync(mStr)
+      .then((_mssgStr) => {
+
+        // message object to be sent to the clients
+        const mssgObj = JSON.parse(_mssgStr);
+        const key = Object.keys(mssgObj)[0];
+
+        /*
+         * pass on the message received through the redis subscriber to the socket
+         * io emitter to send data to the browser clients.
+         */
+        emitter(io, key, mssgObj);
+      });
+    } catch (err) {
+      // message object to be sent to the clients
+      const mssgObj = JSON.parse(mssgStr);
+      const key = Object.keys(mssgObj)[0];
 
-    /*
-     * pass on the message received through the redis subscriber to the socket
-     * io emitter to send data to the browser clients.
-     */
-    emitter(io, key, mssgObj);
+      /*
+       * pass on the message received through the redis subscriber to the socket
+       * io emitter to send data to the browser clients.
+       */
+      emitter(io, key, mssgObj);
+    }
```

Comment:
We can do much better than this but will involve changes in quite a few places. Want to make sure that the compression really helps us before doing the other changes.